src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/SwitchExtensions.cs

```diff
@@ -0,0 +1,254 @@
+﻿using Android.Graphics.Drawables;
+using Xamarin.Forms;
+using ASwitch = AndroidX.AppCompat.Widget.SwitchCompat;
+using AAttribute = Android.Resource.Attribute;
+using Android.Content.Res;
+using AColor = Android.Graphics.Color;
+using System;
+using APorterDuff = Android.Graphics.PorterDuff;
+using AndroidX.Core.Graphics.Drawable;
+using AndroidX.AppCompat.Widget;
+using Android.Content;
+using Android.Util;
+
+namespace Xamarin.Platform
+{
+	public static class SwitchExtensions
+	{
+		static ColorTrackingColorStateList? mSwitchTrackStateList;
+		// taken from android sourcec ccode
+		static ColorTrackingColorStateList getSwitchTrackColorStateList(this Context context)
+		{
+			if (mSwitchTrackStateList == null)
+			{
+				int[][] states = new int[3][];
+				int[] colors = new int[3];
+				int i = 0;
+				// Disabled state
+				states[i] = new int[] { -AAttribute.StateEnabled };
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorForeground, 0.1f);
+				i++;
+				states[i] = new int[] { AAttribute.StateChecked };
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorControlActivated, 0.3f);
+				i++;
+				// Default enabled state
+				states[i] = new int[0];
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorForeground, 0.3f);
+				i++;
+				mSwitchTrackStateList = new ColorTrackingColorStateList(states, colors);
+			}
+			return mSwitchTrackStateList;
+		}
+
+		static int getThemeAttrColor(Context mContext, int attr)
+		{
+			TypedValue mTypedValue = new TypedValue();
+			if (mContext.Theme?.ResolveAttribute(attr, mTypedValue, true) == true)
+			{
+				if (mTypedValue.Type >= DataType.FirstInt
+						&& mTypedValue.Type <= DataType.LastInt)
+				{
+					return mTypedValue.Data;
+				}
+				else if (mTypedValue.Type == DataType.String)
+				{
+					throw new NotImplementedException();
+				}
+			}
+			return 0;
+		}
+
+		static int getThemeAttrColor(Context mContext, int attr, float alpha)
+		{
+			int color = getThemeAttrColor(mContext, attr);
+			int originalAlpha = AColor.GetAlphaComponent(color);
+			// Return the color, multiplying the original alpha by the disabled value
+			return (color & 0x00ffffff) | ((int)Math.Round(originalAlpha * alpha) << 24);
+		}
+
+		static int[][] _checkedStates = new int[][]
+					{
+						new int[] { -AAttribute.StateEnabled,  },
+						new int[] { AAttribute.StateChecked },
+						new int[0],
+					};
+
+
+		public static void UpdateIsToggled(this ASwitch aSwitch, ISwitch view)
+		{
+			aSwitch.Checked = view.IsToggled;
+		}
+
+		//static Drawable? _defaultTrackDrawable;
+		public static void UpdateOnColor(this ASwitch aSwitch, ISwitch view)
```

Comment:
@mattleibow @jsuarezruiz With this route we can just set the ColorStateList and then let it just do its thing.

We no longer have to call into UpdateOnColor every time the check changes